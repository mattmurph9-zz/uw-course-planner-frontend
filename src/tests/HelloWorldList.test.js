import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorldList from '../components/HelloWorldList';
import AddGreeter from '../components/AddGreeter';
import HelloWorld from '../components/HelloWorld';
describe(HelloWorldList, () => {
  const component = shallow(
    <HelloWorldList />
  );
  // Add the rest of our tests here
  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <HelloWorldList />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('contains an AddGreeter subcomponent', () => {
    expect(component.find(AddGreeter)).toHaveLength(1);
  });
  it('contains the same number of HelloWorld components as greetings', () => {
    const helloWorlds = component.find(HelloWorld).length;
    const greetings = component.state('names').length;
    expect(helloWorlds).toEqual(greetings);
  });
  it('adds another greeting when the add greeting function is called', () => {
    const before = component.find(HelloWorld).length;
    component.instance().addGreeting('Test');
    component.update();
    const after = component.find(HelloWorld).length;
    expect(after).toEqual(before + 1);
  });
  it('removes a greeting from the list when the remove greeting function is called', () => {
    const before = component.find(HelloWorld).length;
    const remove = component.state('names')[0];
    component.instance().removeGreeting(remove);
    component.update();
    const after = component.find(HelloWorld).length;
    expect(after).toEqual(before - 1);
  });
});