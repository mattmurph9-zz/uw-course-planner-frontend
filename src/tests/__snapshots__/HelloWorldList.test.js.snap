// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {
    constructor(props) {
        super(props);this.





        renderGreetings = () => {
            return this.state.names.map(name =>
            _react2.default.createElement(_HelloWorld2.default, {
                key: name,
                name: name,
                removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }));


        };this.

        addGreeting = name => {
            this.setState({ names: [...this.state.names, name] });
        };this.

        removeGreeting = removeName => {
            const filteredNames = this.state.names.filter(name => {
                return name !== removeName;
            });
            this.setState({ names: filteredNames });
        };this.state = { names: ['Matt', 'Dan', 'Luke'] };}

    render() {
        return (
            _react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this },
                _react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 39 }, __self: this }),
                this.renderGreetings()));


    }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      className="add"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Matt
    <br />
    <button
      className="changeLang"
      onClick={[Function]}
    >
      Change Language
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Entry
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Dan
    <br />
    <button
      className="changeLang"
      onClick={[Function]}
    >
      Change Language
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Entry
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Luke
    <br />
    <button
      className="changeLang"
      onClick={[Function]}
    >
      Change Language
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Entry
    </button>
  </div>
</div>
`;
